### PR TITLE
Removed dependency of openSSL for authentication with .p12 keys

### PR DESCRIPTION
**Pull request by googlyrahman**
Test Results = https://paste.googleplex.com/5911264932397056
Removed dependency of openSSL for authentication with .p12 keys

### CODE REVIEW

File: gcs_oauth2_boto_plugin/oauth2_client.py [hunk 1]
```diff
@@ -52,6 +52,11 @@
 import oauth2client.service_account
 from google_reauth import reauth_creds
 import retry_decorator.retry_decorator
+from google.auth import _helpers
```
Comment (sushantkhanna) [inline gcs_oauth2_boto_plugin/oauth2_client.py:55]: Should we not bump up the minor version of the library, considering we are changing the underlying auth method ?  
My understanding is that for the user, virtually nothing changes though ? Hence does not require a major version change ?

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,8 @@
 rsa==4.7.2
 boto>=2.29.1
 google-reauth>=0.1.0
+google-auth
```
Comment (sushantkhanna) [inline requirements.txt:4]: Should we specify a version ?

File: gcs_oauth2_boto_plugin/oauth2_client.py (hunk 1 above)
Comment (googlyrahman) [reply gcs_oauth2_boto_plugin/oauth2_client.py:55]: We can do that, that's another process - https://g3doc.corp.google.com/cloud/storage/g3doc/team/sub_teams/serving/sub_teams/orchestration/sub_teams/clients/sub_teams/cli/gsutil/gcs-oauth2-boto-plugin-release.md?cl=head

We can do it as a separate change, our priority currently is to do the release process of gsutil, TODO bug for tracking https://b.corp.google.com/issues/336344983

File: requirements.txt (hunk 2 above)
Comment (googlyrahman) [reply requirements.txt]: Done

**Comment by googlyrahman** [comment]
Closing this, and opening this again to keep the commits clean.
https://github.com/GoogleCloudPlatform/gcs-oauth2-boto-plugin/pull/62

**Pull request closed without merging**